File: lily-backend/app/schemas/incident.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field

from app.utils.enums import Category, IncidentStatus, Severity


class IncidentCreate(BaseModel):
    title: str = Field(min_length=1, max_length=255)
    description: str = Field(min_length=1)
    service: str = Field(min_length=1, max_length=255)
    environment: str = Field(min_length=1, max_length=100)
    source: str = Field(default="api", max_length=255)


class IncidentUpdate(BaseModel):
    status: IncidentStatus | None = None
    severity: Severity | None = None
    category: Category | None = None
    impact: str | None = None
    urgency: str | None = None
    recommended_action: str | None = None


class IncidentResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: uuid.UUID
    title: str
    description: str
    service: str
    environment: str
    severity: Severity
    status: IncidentStatus
    category: Category | None
    source: str
    impact: str | None
    urgency: str | None
    ai_analysis: dict | None
    recommended_action: str | None
    requires_approval: bool
    created_at: datetime
    updated_at: datetime
    resolved_at: datetime | None


class IncidentListResponse(BaseModel):
    items: list[IncidentResponse]
    page: int
    page_size: int
    total: int


class AnalysisResponse(BaseModel):
    incident_id: uuid.UUID
    severity: Severity
    category: Category
    impact: str
    urgency: str
    recommended_action: str
    requires_approval: bool
    reason: str

File: lily-web/src/types/incident.ts
export type IncidentSeverity = 'LOW' | 'MEDIUM' | 'HIGH' | 'CRITICAL'

export type IncidentStatus = 
  | 'OPEN'
  | 'ANALYZING'
  | 'AWAITING_APPROVAL'
  | 'APPROVED'
  | 'IN_PROGRESS'
  | 'RESOLVED'
  | 'FAILED'
  | 'CLOSED'

export type IncidentCategory = 
  | 'PERFORMANCE'
  | 'AVAILABILITY'
  | 'SECURITY'
  | 'DATABASE'
  | 'INFRASTRUCTURE'
  | 'NETWORK'
  | 'APPLICATION'
  | 'OTHER'

export interface AIAnalysis {
  severity: IncidentSeverity
  category: IncidentCategory | string
  confidence?: number // e.g. 0.92 or 92
  confidenceScore?: number
  impact: string
  urgency: string
  summary?: string
  reason?: string
  recommendedAction: string
  recommended_action?: string
  requiresApproval: boolean
  requires_approval?: boolean
}

export interface IncidentTimelineEvent {
  id: string
  title: string
  description?: string
  timestamp: string
  status?: IncidentStatus
  actor?: 'ai' | 'user' | 'system' | 'n8n' | 'fastapi'
  isCurrent?: boolean
  meta?: Record<string, unknown>
}

export interface Incident {
  id: string
  title: string
  description: string
  service: string
  environment: string
  severity: IncidentSeverity
  status: IncidentStatus
  category?: IncidentCategory | string
  source?: string
  impact?: string | null
  urgency?: string | null
  aiConfidence?: number
  aiSummary?: string
  recommendedAction?: string | null
  recommended_action?: string | null
  requiresApproval?: boolean
  requires_approval?: boolean
  aiAnalysis?: AIAnalysis | null
  ai_analysis?: Record<string, unknown> | null
  timeline?: IncidentTimelineEvent[]
  createdAt: string
  created_at?: string
  updatedAt: string
  updated_at?: string
  resolvedAt?: string | null
  resolved_at?: string | null
}

export interface CreateIncidentInput {
  title: string
  description: string
  service: string
  environment: string
  severity?: IncidentSeverity
  source?: string
}

export interface IncidentFilters {
  search?: string
  severity?: IncidentSeverity | 'ALL'
  status?: IncidentStatus | 'ALL'
  service?: string | 'ALL'
  environment?: string | 'ALL'
}
